# origin: torvalds (BitKeeper)
# cset: 1.2199.1.21 (2.6) key=4239dad1BWUxd4WEx388lwZCb05Q0Q
# URL: http://linux.bkbits.net:8080/linux-2.6/cset@4239dad1BWUxd4WEx388lwZCb05Q0Q
# inclusion: upstream
# descrition: isofs: more "corrupted iso image" error cases
# revision date: Mon, 28 Mar 2005 18:47:06 +0900
#
# S rset: ChangeSet|1.2199.1.20..1.2199.1.21
# I rset: fs/isofs/inode.c|1.47..1.48
#
# Key:
# S: Skipped  ChangeSet file only
# O: Original Followed by Updated
# U: Updated  Included with updated range of versions
# I: Included Included verbatim
# E: Excluded Excluded on request from user
# D: Deleted  Manually deleted by subsequent user edit
# R: Revised  Manually revised by subsequent user edit
#
#
# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/03/17 11:30:25-08:00 torvalds@ppc970.osdl.org 
#   isofs: more "corrupted iso image" error cases
#   
#   Thanks to Michal Zalewski for testing.
# 
# fs/isofs/inode.c
#   2005/03/17 11:30:15-08:00 torvalds@ppc970.osdl.org +5 -0
#   isofs: more "corrupted iso image" error cases
#   
#   Thanks to Michal Zalewski for testing.
# 
#
===== fs/isofs/inode.c 1.47 vs 1.48 =====
--- 1.47/fs/isofs/inode.c	2004-08-24 18:08:57 +09:00
+++ 1.48/fs/isofs/inode.c	2005-03-18 04:30:15 +09:00
@@ -685,6 +685,8 @@ root_found:
 	  sbi->s_log_zone_size = isonum_723 (h_pri->logical_block_size);
 	  sbi->s_max_size = isonum_733(h_pri->volume_space_size);
 	} else {
+	  if (!pri)
+	    goto out_freebh;
 	  rootp = (struct iso_directory_record *) pri->root_directory_record;
 	  sbi->s_nzones = isonum_733 (pri->volume_space_size);
 	  sbi->s_log_zone_size = isonum_723 (pri->logical_block_size);
@@ -1394,6 +1396,9 @@ struct inode *isofs_iget(struct super_bl
 	unsigned long hashval;
 	struct inode *inode;
 	struct isofs_iget5_callback_data data;
+
+	if (offset >= 1ul << sb->s_blocksize_bits)
+		return NULL;
 
 	data.block = block;
 	data.offset = offset;
